runtime-modules/content/src/benchmarks/benchmarking.rs

```diff
@@ -0,0 +1,264 @@
+
+        let c in
+        (T::DistributionBucketsPerBagValueConstraint::get().min as u32) ..
+        (T::DistributionBucketsPerBagValueConstraint::get().max() as u32);
+
+        let d in 1 .. MAX_OBJ_NUMBER; //max objs number
+
+        let max_obj_size: u64 = T::MaxDataObjectSize::get();
+
+        let storage_wg_lead_account_id = insert_storage_leader::<T>();
+
+        let distribution_wg_lead_account_id = insert_distribution_leader::<T>();
+
+        let (channel_owner_account_id, channel_owner_member_id) =
+            member_funded_account::<T>(DEFAULT_MEMBER_ID);
+
+        let sender = RawOrigin::Signed(channel_owner_account_id);
+
+        let channel_owner = ChannelOwner::Member(channel_owner_member_id);
+
+        let params = generate_channel_creation_params::<T>(
+            storage_wg_lead_account_id,
+            distribution_wg_lead_account_id,
+            max_obj_size.saturating_mul(MAX_OBJ_NUMBER.into()),
+            MAX_OBJ_NUMBER.into(),
+            a, b, c, d,
+            max_obj_size,
+        );
+
+    }: _ (sender, channel_owner, params.clone())
+    verify {
+
+        let channel_id: T::ChannelId = One::one();
+        assert!(ChannelById::<T>::contains_key(&channel_id));
+
+        let channel = ChannelById::<T>::get(channel_id);
+
+        assert_last_event::<T>(
+            Event::<T>::ChannelCreated(
+                channel_id,
+                channel,
+                params).into()
+        );
+    }
+
+    update_channel {
+
+        let a in 1 .. MAX_COLABORATOR_IDS as u32; //max colaborators
+
+        let b in 1 .. MAX_OBJ_NUMBER; //max objs number to upload
+
+        let c in 1 .. MAX_OBJ_NUMBER; //max objs number to remove
+
+        let max_obj_size: u64 = T::MaxDataObjectSize::get();
+
+        let storage_wg_lead_account_id = insert_storage_leader::<T>();
+
+        let distribution_wg_lead_account_id = insert_distribution_leader::<T>();
+
+        let (channel_owner_account_id, channel_owner_member_id) =
+            member_funded_account::<T>(DEFAULT_MEMBER_ID);
+
+        let sender = RawOrigin::Signed(channel_owner_account_id.clone());
+
+        let channel_owner =
+            ChannelOwner::Member(channel_owner_member_id);
+
+        let assets_to_remove =
+            Storage::<T>::get_next_data_object_ids(c as usize);
+
+        let params = generate_channel_creation_params::<T>(
+            storage_wg_lead_account_id,
+            distribution_wg_lead_account_id,
+            max_obj_size.saturating_mul((2 * MAX_OBJ_NUMBER).into()),
+            (2 * MAX_OBJ_NUMBER).into(),
+            a,
+            T::StorageBucketsPerBagValueConstraint::get().max() as u32,
+            T::DistributionBucketsPerBagValueConstraint::get().max() as u32,
+            c,
+            max_obj_size,
+        );
+
+        Pallet::<T>::create_channel(
+            sender.into(),
+            channel_owner,
+            params.clone()).unwrap();
+
+        let permissions = BTreeSet::from_iter([
```

Comment:
I'm not sure how the size of this set will affect the weight, but probably worth using either all possible variants or all minus one (in order to make them different from the current permissions)

Reply:
The previous review comment is still not addressed. I suggest using a set like:
```suggestion
        let permissions: ChannelAgentPermissions = worst_case_channel_agent_permissions().into_iter().skip(1).collect();
```